### PR TITLE
refactor: ampd configs should contain a Vec of handler configs

### PR DESCRIPTION
**Pull request by sammy1991106**
## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: ampd/src/handlers/config.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+#[derive(Debug, Deserialize)]
+pub struct Chain {
+    pub name: ChainName,
+    pub rpc_url: Url,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type")]
+pub enum Config {
+    EVMMsgVerifier {
```
Comment (cgorenflo) [inline ampd/src/handlers/config.rs:18]:
```suggestion
    EvmMsgVerifier {
```

File: ampd/src/handlers/config.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+fn validate_evm_msg_verifier_configs<'de, D>(configs: &[Config]) -> Result<(), D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let evm_msg_verifier_configs: Vec<_> = configs
+        .iter()
+        .filter(|config| matches!(config, Config::EVMMsgVerifier { .. }))
+        .collect();
```
Comment (cgorenflo) [inline ampd/src/handlers/config.rs:50]: filter_map on chain here already? Then you just need to convert it into a HashSet in the next step

File: ampd/src/handlers/config.rs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+where
+    D: Deserializer<'de>,
+{
+    let configs: Vec<Config> = Deserialize::deserialize(deserializer)?;
+
+    validate_evm_msg_verifier_configs::<D>(&configs)?;
+    validate_multisig_signer_config::<D>(&configs)?;
+
+    Ok(configs)
+}
```
Comment (cgorenflo) [inline ampd/src/handlers/config.rs:84]: love it

File: ampd/src/handlers/config.rs (hunk 2 above)
Comment (sammy1991106) [reply ampd/src/handlers/config.rs:50]: hmm let me try if I can make it clearer. The filter part of the `filter_map` below isn't really doing anything.

File: ampd/Cargo.toml [hunk 4]
```diff
@@ -25,6 +25,7 @@ events-derive = { workspace = true }
 futures = "0.3.25"
 hex = { version = "0.4.3", features = ["serde"] }
 humantime-serde = "1.1.1"
+itertools = "0.11.0"
```
Comment (sammy1991106) [inline ampd/Cargo.toml:28]: @cgorenflo Maybe it's worth pulling this into a workspace dependency?

**Review by cgorenflo — APPROVED** [review]

**Review by haiyizxx — APPROVED** [review]

**Pull request merged**